.github/workflows/devcontainer.yaml

```diff
@@ -0,0 +1,53 @@
+name: devcontainer
+
+on:
+  push:
+    paths:
+      - .devcontainer/base/Dockerfile
+      - .devcontainer/base/devcontainer.json
+  pull_request:
+    paths:
+      - .devcontainer/base/Dockerfile
+      - .devcontainer/base/devcontainer.json
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - id: set-platforms
+        run: |
+          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
+            echo "platforms=linux/amd64,linux/arm64" >>$GITHUB_OUTPUT
+          else
+            echo "platforms=linux/amd64" >>$GITHUB_OUTPUT
+          fi
+
+      - name: Pre-build dev container image
+        uses: devcontainers/ci@v0.2
+        with:
+          subFolder: .devcontainer/base
+          imageName: ghcr.io/prql/prql-devcontainer
+          cacheFrom: ghcr.io/prql/prql-devcontainer
```

Comment:
> it's much easier than this for the basic Docker caching — this does the basic build cache already, using the image repository:

Inline cache is used here.

Reply:
Yes great! That's all the caching we need; no need for the build mount